File: java/src/main/java/org/shypl/sna/impl/FbSocialNetwork.java
package org.shypl.sna.impl;

import org.shypl.common.util.DeclinationImpl;
import org.shypl.sna.Currency;
import org.shypl.sna.SocialNetwork;

public final class FbSocialNetwork extends SocialNetwork {
	public FbSocialNetwork() {
		super(4, "fb", "Facebook", new Currency("fb_currency", new DeclinationImpl("fb_currency", "fb_currency", "fb_currency")));
	}
}

File: php/src/main/php/org/shypl/sna/impl/FbSocialNetwork.php
<?php

namespace org\shypl\sna\impl;

use org\shypl\sna\Adapter;
use org\shypl\sna\SocialNetwork;

final class FbSocialNetwork extends SocialNetwork {
	const ID = 4;

	/**
	 *
	 */
	public function __construct() {
		parent::__construct(self::ID, "fb");
	}

	/**
	 * @param array $parameters
	 *
	 * @return Adapter
	 */
	public function createAdapter(array $parameters) {
		return new FbAdapter($parameters);
	}
}